components/scream/src/physics/spa/spa_functions_impl.hpp

```diff
@@ -0,0 +1,294 @@
+      const auto& aer_g_sw_src_sub           = ekat::subview(aer_g_sw_src, i, n);
+
+      const auto& aer_ssa_sw_beg_sub         = ekat::subview(data_beg.AER_SSA_SW, i, n);
+      const auto& aer_ssa_sw_end_sub         = ekat::subview(data_end.AER_SSA_SW, i, n);
+      const auto& aer_ssa_sw_src_sub         = ekat::subview(aer_ssa_sw_src, i, n);
+  
+      const auto& aer_tau_sw_beg_sub         = ekat::subview(data_beg.AER_TAU_SW, i, n);
+      const auto& aer_tau_sw_end_sub         = ekat::subview(data_end.AER_TAU_SW, i, n);
+      const auto& aer_tau_sw_src_sub         = ekat::subview(aer_tau_sw_src, i, n);
+
+      /* Now loop over fastest index, the number of vertical packs */
+      Kokkos::parallel_for(
+        Kokkos::ThreadVectorRange(team, nk_pack), [&] (Int k) {
+        aer_g_sw_src_sub(k)   = linear_interp(aer_g_sw_beg_sub(k),   aer_g_sw_end_sub(k),   t_norm);
+        aer_ssa_sw_src_sub(k) = linear_interp(aer_ssa_sw_beg_sub(k), aer_ssa_sw_end_sub(k), t_norm);
+        aer_tau_sw_src_sub(k) = linear_interp(aer_tau_sw_beg_sub(k), aer_tau_sw_end_sub(k), t_norm);
+      });
+    });
+    team.team_barrier();
+    }
+    {
+    /* Loop over all LW variables with nlwbands */
+    Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, nlwbands), [&] (int n) {
+      const auto& aer_tau_lw_beg_sub         = ekat::subview(data_beg.AER_TAU_LW, i, n);
+      const auto& aer_tau_lw_end_sub         = ekat::subview(data_end.AER_TAU_LW, i, n);
+      const auto& aer_tau_lw_src_sub         = ekat::subview(aer_tau_lw_src, i, n);
+
+      /* Now loop over fastest index, the number of vertical packs */
+      Kokkos::parallel_for(
+        Kokkos::ThreadVectorRange(team, nk_pack), [&] (Int k) {
+        aer_tau_lw_src_sub(k) = linear_interp(aer_tau_lw_beg_sub(k), aer_tau_lw_end_sub(k), t_norm);
+      });
+    });
+    team.team_barrier();
+    }
+  });
+  Kokkos::fence();
+
+  // Third Step: Vertical interpolation, project the SPA data onto the pressure profile for this simulation.
+  // This is done using the EKAT linear interpolation routine, see /externals/ekat/util/ekat_lin_interp.hpp
+  // for more details. 
+  using LIV = ekat::LinInterp<Real,Spack::n>;
+  Real minthreshold = 0.0;  // Hard-code a minimum value for aerosol concentration to zero.
+
+  LIV VertInterp(ncols_atm,pressure_state.nlevs,nlevs_atm,minthreshold);
+  /* Parallel loop strategy:
+   * 1. Loop over all simulation columns (i index)
+   * 2. Where applicable, loop over all aerosol bands (n index)
+   */ 
+  const Int most_bands = std::max(nlwbands, nswbands);
+  typename LIV::TeamPolicy band_policy(ncols_atm, ekat::OnGpu<typename LIV::ExeSpace>::value ? most_bands : 1, VertInterp.km2_pack());
+  Kokkos::parallel_for("vertical-interp-spa",
+    band_policy,
+    KOKKOS_LAMBDA(typename LIV::MemberType const& team) {
+    const int i = team.league_rank();
+    /* Setup the linear interpolater for this column. */
+    VertInterp.setup(team,
```

Comment:
This call to `setup` and the following one to `lin_interp` will dispatch a TTR, but given that the vector length is greater than 0, there will be 32 threads running the setup/lin_interp at the same time. I think everything "works", since both `setup` and `lin_interp` don't contain lines like `some_view(k) += blah`, but always do += on thread-local variables, and always use plain assignment when it's time to write into the view. So every thread will write on the output views, but they will all write the same thing.

That said, I _think_ this is not good programming. As soon as someone makes a tiny change to setup/lin_interp, all the code that used them with TTR boundaries, but with a policy with vector length>1 could start failing, and the bug would be _very hard_ to find. Furthermore, 32 threads all trying to write at the same memory location is not very performant (not coalesced access).

I suggest doing somethign like this:
```
Kokkos::single(Kokkos::PerTeam(team),[&](){
  const auto tvr = Kokkos::ThreadVectorRange(team, VertInterp.km2_pack());
  VectInterp.setup(team,tvr,...);
});
```
and similarly for the 2d lin_interp call.